### PR TITLE
test: include agent metrics in dump

### PR DESCRIPTION
**Pull request by howardjohn**
15020 includes envoy+agent, 15090 is just envoy

### CODE REVIEW

File: pkg/test/kube/dump.go [hunk 1]
```diff
@@ -440,9 +440,9 @@ func DumpPodProxies(ctx resource.Context, c cluster.Cluster, workDir, namespace
 				return err
 			}
 			defer fw.Close()
+			defer fw.Close()
```
Comment (dhawton) [inline pkg/test/kube/dump.go:443]: Is this meant to be duplicated?

Comment (howardjohn) [reply pkg/test/kube/dump.go:443]: nope, that is why the test is failing. will fix. Thanks!

**Review by dhawton — APPROVED** [review]
LGTM

**Pull request merged**